src/main/java/com/microsoft/azure/vmagent/AzureVMCloud.java

```diff
@@ -471,52 +477,49 @@ public AzureVMAgent createProvisionedAgent(
-            PagedList<Deployment> deployments= azureClient.deployments().listByGroup(resourceGroupName);
-            for (Deployment dep : deployments) {
-                PagedList<DeploymentOperation> ops = dep.deploymentOperations().list();
-                for (DeploymentOperation op : ops) {
-                    if (op.targetResource() == null) {
-                        continue;
-                    }
-                    final String resource = op.targetResource().resourceName();
-                    final String type = op.targetResource().resourceType();
-                    final String state = op.provisioningState();
-                    if (op.targetResource().resourceType().contains("virtualMachine")) {
-                        if(resource.equalsIgnoreCase(vmName)) {
-
-                            if (!state.equalsIgnoreCase("creating")
-                                && !state.equalsIgnoreCase("succeeded")
-                                && !state.equalsIgnoreCase("running")){
-                                final String statusCode = op.statusCode();
-                                final Object statusMessage = op.statusMessage();
-                                String finalStatusMessage = statusCode;
-                                if (statusMessage != null) {
-                                    finalStatusMessage += " - " + statusMessage.toString();
-                                }
-                                throw new AzureCloudException(String.format("AzureVMCloud: createProvisionedAgent: Deployment %s: %s:%s - %s", new Object[]{state, type, resource, finalStatusMessage}));
-                            }
-                            else if (state.equalsIgnoreCase("succeeded")) {
-                                LOGGER.log(Level.INFO, "AzureVMCloud: createProvisionedAgent: VM available: {0}", resource);
-
-                                final VirtualMachine vm = azureClient.virtualMachines().getByGroup(resourceGroupName, resource);
-                                final OperatingSystemTypes osType = vm.storageProfile().osDisk().osType();
-                                
-                                AzureVMAgent newAgent = AzureVMManagementServiceDelegate.parseResponse(vmName, deploymentName, template, osType);
-                                AzureVMManagementServiceDelegate.setVirtualMachineDetails(newAgent, template);
-                                return newAgent;
-                            } else {
-                                LOGGER.log(Level.INFO, "AzureVMCloud: createProvisionedAgent: Deployment {0} not yet finished ({1}): {2}:{3} - waited {4} seconds",
-                                    new Object[]{deploymentName, state, type, resource,
-                                        (maxTries - triesLeft) * sleepTimeInSeconds});
+            Deployment dep = azureClient.deployments().getByName(deploymentName);
+            PagedList<DeploymentOperation> ops = dep.deploymentOperations().list();
```

Comment:
check for null:
http://devops-ci.westcentralus.cloudapp.azure.com/job/azure-vm-agents/job/mmitche/job/various-fixes/1/testReport/junit/com.microsoft.azure.vmagent.test/ITAzureVMCloud/createProvisionedAgentThowsExceptionWhenNoDeployments/